rhobs/alerting/data_plane/prometheus.release_service_alerts.yaml

```diff
@@ -0,0 +1,77 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: rhtap-release-service-alerting
+  labels:
+    tenant: rhtap
+spec:
+  groups:
+  - name: release_service_alerts
+    interval: 1m
+
+    rules:
+    # record rules for `ReleaseServiceProcessingDuration`
+    - record: rate_of_release_processing_duration_seconds_longer_than_1h_per_10m
+      expr: sum by (job) (rate(release_processing_duration_seconds_bucket{le="3600"}[10m]))
+
+    - record: rate_of_release_processing_duration_seconds_counter_per_10m
+      expr: sum by (job) (rate(release_processing_duration_seconds_count[10m]) > 0)
+
+    # record rules for `ReleaseServicePreProcessingDurationSeconds`
+    - record: rate_of_release_pre_processing_duration_seconds_longer_than_10s_per_10m
+      expr: sum by (job) (rate(release_pre_processing_duration_seconds_bucket{le="10"}[10m]))
+
+    - record: rate_of_release_pre_processing_duration_seconds_counter_per_10m
+      expr: sum by (job) (rate(release_pre_processing_duration_seconds_count[10m]) > 0)
+
+    # record rules for `ReleaseServiceValidationDurationSeconds`
+    - record: rate_of_release_validation_duration_seconds_longer_than_5s_per_10m
+      expr: sum by (job) (rate(release_validation_duration_seconds_bucket{le="5"}[10m]))
+
+    - record: rate_of_release_validation_duration_seconds_counter_per_10m
+      expr: sum by (job) (rate(release_validation_duration_seconds_count[10m]) > 0)
+
+    - alert: ReleaseServiceProcessingDuration
+      expr: |
+        (
+          rate_of_release_processing_duration_seconds_longer_than_1h_per_10m / rate_of_release_processing_duration_seconds_counter_per_10m
+        ) < 0.90
```

Comment:
Is this the correct operator? We want to alert when this ratio is larger than 0.9. No?

Reply:
In this case it is the other way around. It should alert if the majority of the cases goes above the threshold. 
Correct me if my the logic does not represent the calculation:

We have 100 releases and the majority should fall in the buckets smaller than 3600. If get reach 0.90 of the requests falling in the bucket 3600, I want to alert.

Reply:
Maybe the recording rule should be named `rate_of_release_processing_duration_seconds_shorter_than_1h_per_10m` instead of `...longer_than...`.

Reply:
Reading now, my last phrase was unclear actually. Is not  "reach" 0.90, but "fall below" 0.90.
The healthy state is between `0.90` and `1.0`. But I see -  although I am searching for pipelineruns in the "+3600" bucket, the result regards to the ones below it. I will change.